### PR TITLE
bug(web): map overlays blocking touch events on mobile

### PR DESCRIPTION
**Pull request by tonypls**
## Issue

Empty map overlay component is blocking touch events on smaller devices

## Description

This PR hides the component on smaller devices

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
Not sure if this is just the micro surveys doing but it seems like this entire area is blocking all the map interaction (on both desktop and mobile) likely due to the fact that the micro survey is a lot wider than the others.

image

Also should the legend really move? Is that not confusing and causing undue layout shifts and renders?

File: web/src/components/MapOverlays.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default function MapOverlays() {
   const surveyEnabled = useFeatureFlag('feedback-micro-survey') && !hasSeenSurveyCard;
 
   return (
-    <div className="fixed bottom-4 right-4 z-20 flex flex-col items-end space-y-3">
+    <div className="fixed bottom-4 right-4 z-20 hidden flex-col items-end space-y-3 sm:flex">
       <Suspense>
         <FeatureFlagsManager />
       </Suspense>
```
Comment (VIKTORVAV99) [inline web/src/components/MapOverlays.tsx:18]: I think we need to wrap each of these in their own div so they only take up the exact width and high each of the elements require.

**Comment by tonypls** [comment]
> Not sure if this is just the micro surveys doing but it seems like this entire area is blocking all the map interaction (on both desktop and mobile) likely due to the fact that the micro survey is a lot wider than the others.
> 
> image
> Also should the legend really move? Is that not confusing and causing undue layout shifts and renders?

I updated the clicks to only propagate where required. I think it's a bit nicer for users that it renders below the legend as it's a common place for feedback cards. 

Let's focus on the bug here and we can think about UX improvements after if we like 👨‍🎨

File: web/src/components/MapOverlays.tsx (hunk 1 above)
Comment (tonypls) [reply web/src/components/MapOverlays.tsx:18]: I'm not sure this will help 🤔

Comment (VIKTORVAV99) [reply web/src/components/MapOverlays.tsx:18]: Wouldn't wrapping each of them in their individual divs ensure they don't cover any more of the screen than they should? Right now it forms a big (transparent) box around the extreme points.

Comment (tonypls) [reply web/src/components/MapOverlays.tsx:18]: I tried wrapping in divs when you suggested and it still grew to fill the larger container

**Review by VIKTORVAV99 — APPROVED** [review]
LGMT! 🎉

File: web/src/components/MapOverlays.tsx (hunk 1 above)
Comment (VIKTORVAV99) [reply web/src/components/MapOverlays.tsx:18]: Ah, that's unfortunate. Let's go with the current approach then.

**Pull request merged**